### PR TITLE
Affix - Allow for using verticalAlignment with top and bottom positioning

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment](https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment](https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment](https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment](https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment)

File: src/app/public/modules/affix/affixer.ts [hunk 1]
```diff
@@ -248,8 +247,32 @@ export class SkyAffixer {
...
+        switch (verticalAlignment) {
+          case 'top':
+            top = top + affixedRect.height;
+            break;
+          case 'middle':
+            top = top + (affixedRect.height / 2);
+            break;
+          case 'bottom':
+            default:
+              break;
```
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/affix/affixer.ts]: Fixed. I had noticed this but an existing case in the file had this spacing so I had just followed that precedence (I don't write switch cases very often - lol). I fixed this and also fixed the existing one.

File: src/app/public/modules/affix/affixer.ts [hunk 2]
```diff
@@ -248,8 +247,32 @@ export class SkyAffixer {
...
+        }
       } else {
         top = baseRect.bottom;
+
+        switch (verticalAlignment) {
+          case 'top':
+            default:
+              break;
```
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/affix/affixer.ts]: Fixed

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment](https://developer.blackbaud.com/skyux-core?_sp.spa=pr-affix-top-and-bottom-vertical-alignment)

**Pull request merged**